src/plugins/intel_gpu/src/runtime/device.cpp

```diff
@@ -4,64 +4,128 @@
+        }
+
+        if (gfx_version_list.size() == 1) {
+            return info.gfx_ver == gfx_version_list[0];
+        } else if (gfx_version_list.size() == 2) {
+            return info.gfx_ver >= gfx_version_list[0] && info.gfx_ver <= gfx_version_list[1];
         }
-        break;
+        return false;
     }
-    case cldnn::data_types::f16: {
+
+    float get_ops(device_info& info, data_types dt) const {
         if (info.supports_immad) {
-            if (info.gfx_ver.major == 12) {
-                if (info.gfx_ver.minor == 5)
-                    opsPerComputeBlock = 256;
-                else if (info.gfx_ver.minor == 7)
-                    opsPerComputeBlock = 128;
+            auto it = ops_dp4a.find(dt);
+            if (it != ops_dp4a.end()) {
+                return it->second;
             }
+        }
+
+        if (info.supports_imad) {
+            auto it = ops_dpas.find(dt);
+            if (it != ops_dpas.end()) {
+                return it->second;
+            }
+        }
+
+        return get_ops_for_mad(dt);
+    }
+
+    float get_ops_for_mad(data_types dt) const {
+        auto it = ops_mad.find(dt);
+        if (it != ops_mad.end()) {
+            return it->second;
         } else {
-            // fma * simd size
-            opsPerComputeBlock = 2 * 16;
+            return 0;
         }
-        break;
     }
-    case cldnn::data_types::f32: {
-        // fma * simd size
-        opsPerComputeBlock = 2 * 8;
-        break;
+};
+
+const uint8_t MAX_REVISION = 0xff;
+
+const std::vector<DeviceOps> device_ops_table = {
+//    | platform | device_id | gfx_ver                                 | MAD                   | DPAS (imad)           | DP4A (immad)  |
+//    |          |           |                                         |  fp64 |  fp32 |  fp16 |  fp16 |  int8 |  int4 |  fp16 |  int8 |
+    { "Legacy",   {},         { { 0,  0,  0}, {12,  9, MAX_REVISION} }, {     0,     16,     32,      0,      0,      0,      0,      0 } },    // TGL, ADL-S
+    { "DG1",      {},         { {12, 10,  0}                         }, {     0,     16,     32,      0,      0,      0,      0,      0 } },
+    { "DG2",      {},         { {12, 55,  0}, {12, 57, MAX_REVISION} }, {     0,     16,     32,      0,      0,      0,    128,    256 } },
+    { "PVC_XL",   {},         { {12, 60,  0}, {12, 60, 1}            }, {    16,     32,     64,      0,      0,      0,    512,   1024 } },
+    { "PVC_XT",   {},         { {12, 60,  3}, {12, 61, 7}            }, {    32,     32,     64,      0,      0,      0,    512,   1024 } },
+    { "MTL/ARL",  {},         { {12, 70,  0}, {12, 74, MAX_REVISION} }, {   0.5,     16,     32,    128,    256,      0,      0,     64 } },
+    { "ATS",      { 0x020A, 0x0210 },
+                              {},                                       {     0,     16,     32,    128,    256,    512,      0,      0 } },
```

Comment:
This table is confusing
* let's remove int4 from the table because we don't support it
* Why ATS(values on imad) has different value compared to DG2(values on immad)?

Reply:
Misread the spec; this was from the ATS-P of gfx spec. and will be removed.